app/authentication/auth.py

```diff
@@ -47,7 +47,9 @@ def requires_no_auth():
 def restrict_ip_sms():
     ip = ''
     if request.headers.getlist("X-Forwarded-For"):
-        ip = request.headers.getlist("X-Forwarded-For")[0]
+        ip0 = request.headers.getlist("X-Forwarded-For")
+        ip1 = ip0.split(',')
```

Comment:
This won't work - getlist will return a list, and you can't call `split` on lists.

I'd recommend that you write some unit tests for this function, it's starting to get complex.